ethers-etherscan/src/lib.rs

```diff
@@ -59,8 +59,11 @@ impl Cache {
 
     fn set<T: Serialize>(&self, prefix: &str, address: Address, item: T) -> Result<()> {
         let path = self.0.join(prefix).join(format!("{:?}.json", address));
-        let writer = std::io::BufWriter::new(std::fs::File::create(path)?);
-        serde_json::to_writer(writer, &item)?;
+        let mut writer = std::io::BufWriter::new(std::fs::File::create(path)?);
+        serde_json::to_writer(&mut writer, &item)?;
+        // TODO: Trace
+        // TODO: Should we cache if the contract is *not* verified?
```

Comment:
I think not?

Reply:
You could maybe add a TTL? Like only check the cache once a week, and (maybe) add a force flag to bypass the TTL?